apigw-s3-terraform/README.md

```diff
@@ -0,0 +1,84 @@
+# REST API as an Amazon S3 proxy in API Gateway
+
+This pattern in CDK offers a boilerplate to generate an Amazon API Gateway REST API endpoint to Amazon S3 with the following methods -
+* Expose GET on the API's root resource to list all of the Amazon S3 buckets of a caller
+* Expose GET on a Folder resource to view a list of all of the objects in an Amazon S3 bucket
+* Expose GET on a Folder/Item resource to view or download an object from an Amazon S3 bucket
+* Expose PUT on a Folder/Item resource to upload an object to an Amazon S3 bucket
+* Expose HEAD on a Folder/Item resource to get object metadata in an Amazon S3 bucket 
+
+## Requirements
+
+* [Create an AWS account](https://portal.aws.amazon.com/gp/aws/developer/registration/index.html) if you do not already have one and log in. The IAM user that you use must have sufficient permissions to make necessary AWS service calls and manage AWS resources.
+* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) installed and configured
+* [Git Installed](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+* [Node and NPM](https://nodejs.org/en/download/) installed
+* [AWS Cloud Development Kit](https://docs.aws.amazon.com/cdk/latest/guide/cli.html) (AWS CDK) installed
+
+## Deployment Instructions
+
+1. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
+    ``` 
+    git clone https://github.com/aws-samples/serverless-patterns
+    ```
+1. Change directory to the pattern directory:
+    ```
+    cd apigw-s3-terraform
+    ```
+1. From the command line, initialize terraform to  to downloads and installs the providers defined in the configuration:
+    ```
+    terraform init
+    ```
+1. From the command line, apply the configuration in the main.tf file:
+    ```
+    terraform apply
```

Comment:
Getting the following error on terraform apply

│ Error: Invalid provider configuration
│ 
│ Provider "registry.terraform.io/hashicorp/aws" requires explicit configuration. Add a provider block to the root module and configure the provider's required arguments as described in the provider documentation.
│ 
╵
╷
│ Error: Invalid AWS Region: yes
│ 
│   with provider["registry.terraform.io/hashicorp/aws"],
│   on <input-prompt> line 1:
│   (source code not available)
│

Reply:
Changes applied on new main.tf. Please, could you check?